client/v2/autocli/flag/binary.go

```diff
@@ -0,0 +1,65 @@
+package flag
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"os"
+	"path/filepath"
+
+	"github.com/cockroachdb/errors"
+)
+
+type binaryType struct{}
+
+func (f binaryType) NewValue(_ context.Context, _ *Builder) Value {
+	return &fileBinaryValue{}
+}
+
+func (f binaryType) DefaultValue() string {
+	return ""
+}
+
+// fileBinaryValue is a Value that holds a binary file.
+type fileBinaryValue struct {
+	value []byte
+}
+
+func (f *fileBinaryValue) Get(protoreflect.Value) (protoreflect.Value, error) {
+	return protoreflect.ValueOfBytes(f.value), nil
+}
+
+func (f *fileBinaryValue) String() string {
+	return string(f.value)
+}
+
+// Set implements the flag.Value interface for binary files.
+//If the input string is a valid file path the file path most have a file extension, it will read the file.
+//If the input string is a valid hex or base64 encoded string, it will decode the string.
+//If the input string is neither a valid file path, hex, or base64 encoded,
+// it will return an error.
+func (f *fileBinaryValue) Set(s string) error {
+	var value []byte
+	var err error
+
+	if _, err = os.Stat(s); err == nil {
```

Comment:
I think we should check if the input contains "." and attempt reading the file only then.
Otherwise having a file without an extension somewhere may prohibit you starting the program.